core/server/lib/image/blog-icon.js

```diff
@@ -2,121 +2,122 @@ const sizeOf = require('image-size');
 const Promise = require('bluebird');
 const _ = require('lodash');
 const path = require('path');
-const config = require('../../../shared/config');
-const {i18n} = require('../common');
 const errors = require('@tryghost/errors');
-const urlUtils = require('../../../shared/url-utils');
-const settingsCache = require('../../services/settings/cache');
-const storageUtils = require('../../adapters/storage/utils');
 
-/**
- * Get dimensions for ico file from its real file storage path
- * Always returns {object} getIconDimensions
- * @param {string} path
- * @returns {Promise<Object>} getIconDimensions
- * @description Takes a file path and returns ico width and height.
- */
-const getIconDimensions = function getIconDimensions(storagePath) {
-    return new Promise(function getIconSize(resolve, reject) {
-        let dimensions;
-
-        try {
-            dimensions = sizeOf(storagePath);
+class BlogIcon {
+    constructor({config, i18n, urlUtils, settingsCache, storageUtils}) {
+        this.config = config;
+        this.i18n = i18n;
+        this.urlUtils = urlUtils;
+        this.settingsCache = settingsCache;
+        this.storageUtils = storageUtils;
+    }
 
-            if (dimensions.images) {
-                dimensions.width = _.maxBy(dimensions.images, function (w) {
-                    return w.width;
-                }).width;
-                dimensions.height = _.maxBy(dimensions.images, function (h) {
-                    return h.height;
-                }).height;
+    /**
+     * Get dimensions for ico file from its real file storage path
+     * Always returns {object} getIconDimensions
+     * @param {string} path
+     * @returns {Promise<Object>} getIconDimensions
+     * @description Takes a file path and returns ico width and height.
+     */
+    getIconDimensions(storagePath) {
```

Comment:
Think making this method `async` would make sense? makes it slightly more explicit imo and no need to do the `new Promise((resolve, reject) => {` wrapper ;)

Reply:
I'm a bit lost as to why we're using a Promise here. I've just moved the function to being a class method so I didn't dive into it.
I think it would be best to get rid of the promise completely given that there is no async steps here, I'll do that in a follow-up PR.

Reply:
yup, getting rid of unneeded async functions would be great. might be that the function evolved over time and just lock the need to do async operations